include/hobbes/storage.H

```diff
@@ -1437,7 +1437,9 @@ template <size_t n, typename ... Ts>
 template <size_t tag, typename T, typename M>
   struct variantGenWrite {
     static bool fn(T* vd, wpipe* f) {
-      return store<T>::write(*f, *vd);
+      if(store<size_t>::write(*f, tag))
```

Comment:
Nice find!  I'm surprised that this one slipped through.

One small issue is that the decoder will expect the variant tag to be written as a 32-bit int (not size_t).